shark-graph/src/main/java/shark/HprofHeapGraph.kt

```diff
@@ -103,13 +103,18 @@ class HprofHeapGraph internal constructor(
   // Pixel 2 XL API 28. Hit count was ~120K, miss count ~290K
   private val objectCache = LruCache<Long, ObjectRecord>(3000)
 
+  // java.lang.Object is the most accessed class in Heap, so we want to memoize a reference to it
+  private val javaLangObjectClass: HeapClass? = findClassByName("java.lang.Object")
+
   override fun findObjectById(objectId: Long): HeapObject {
     return findObjectByIdOrNull(objectId) ?: throw IllegalArgumentException(
         "Object id $objectId not found in heap dump."
     )
   }
 
   override fun findObjectByIdOrNull(objectId: Long): HeapObject? {
+    if (objectId == javaLangObjectClass?.objectId) return javaLangObjectClass
```

Comment:
Doesn't `javaLangObjectClass?.objectId` return a `Long?` and therefore a wrapper? if yes, we should move the code around.

Reply:
Valid concern! 
Kotlin compiler is pretty smart about redundant allocations and instead of returning a wrapped `Long` it wraps the check if additional `if (value != null)`. 
Here's the decompiled code of the method:
```
@Nullable
public HeapObject findObjectByIdOrNull(long objectId) {
   HeapClass var10001 = this.javaLangObjectClass;
   if (var10001 != null) {
      if (objectId == var10001.getObjectId()) {
         return (HeapObject)this.javaLangObjectClass;
      }
   }
   IndexedObject var10000 = this.index.indexedObjectOrNull(objectId);
   if (var10000 != null) {
      IndexedObject indexedObject = var10000;
      return this.wrapIndexedObject(indexedObject, objectId);
   } else {
      return null;
   }
}
```
And the `HeapClass` keeps the objectId (and its getter) as `long`:
```
private final long objectId;
public long getObjectId() {
   return this.objectId;
}
```

So, based on the code, there's no allocations happening. Still, I wanted to double-check it, so I looked at the allocations when method is invoked and there's no `Long` instances being created 
![image](https://user-images.githubusercontent.com/2152942/85208077-2d98b100-b2f3-11ea-8223-732c2d87d8a5.png)

We're good here :)

Reply:
Nope:

```
   public HeapObject findObjectByIdOrNull(long objectId) {
      HeapClass var10001 = this.javaLangObjectClass;
      if (var10001 != null) {
         if (objectId == var10001.getObjectId()) {
            return (HeapObject)this.javaLangObjectClass;
         }
      }
```

👍